### PR TITLE
fix(Scripts/Underbog): Moved Gha'zan script to C++ file.

### LINKED ISSUE #13793: [The Underbog] Ghaz'an missing scripted waypoints
### Current Behaviour

With the Underbog reconstruction (https://github.com/azerothcore/azerothcore-wotlk/pull/13769) spawns on top of the platform with wander_distance of 12

### Expected Blizzlike Behaviour

Spawns in the lake below the platform, swimming in a circular pattern. Once atreatrigger 4302 is reached, it starts moving towards the platform. Once this waypoint is done, it will wander with a range of 12

### Source

WotLK Classic sniffs

### Steps to reproduce the problem

`.go xyz 251.88 -434.55 81.372864 546 4.6`
Note where Ghaz'an spawns, see if it is swimming in a circle
go to areatrigger 4302 (`.go xyz 241.09 -373.76 72.447418 546 3.75`)
See if it moves towards the platform
When it reaches the platform, see if it wanders

### Extra Notes

Please use my Underbog PR if it isn't merged :) https://github.com/azerothcore/azerothcore-wotlk/pull/13769

Parsed spawn point
```
INSERT INTO `creature` (`guid`, `id1`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `MovementType`, `wander_distance`, `curhealth`, `curmana`, `currentwaypoint`, `npcflag`, `unit_flags`, `dynamicflags`, `VerifiedBuild`) VALUES
(@CGUID+92 , 18105, 546, 3716, 3716, 3, 1, 0, 193.6833953857421875, -425.00103759765625, 43.53683853149414062, 4.910074710845947265, 86400, 0, 0, 59140, 0, 0, 0, 0, 0, 46368), -- 18105 (Area: 3716 - Difficulty: 1) (Auras: 8876 - 8876) (possible waypoints or random movement)
```

Waypoints while swimming in the lake
```
[0] [0] Points: X: 275.60797 Y: -469.67816 Z: 33.473022
[0] [1] Points: X: 276.92447 Y: -457.9421 Z: 37.056316
[0] [2] Points: X: 278.24097 Y: -446.20605 Z: 40.63961
[0] [3] Points: X: 260.68362 Y: -421.5021 Z: 43.44514
[0] [4] Points: X: 233.34468 Y: -409.80292 Z: 45.111805
[0] [5] Points: X: 193.74 Y: -423.40268 Z: 43.583996
[0] [6] Points: X: 220.28748 Y: -447.8345 Z: 43.722916
[0] [7] Points: X: 217.26556 Y: -424.75412 Z: 45.52847
[0] [8] Points: X: 225.28162 Y: -398.65262 Z: 45.4729
[0] [9] Points: X: 253.49117 Y: -397.8943 Z: 45.52849
[0] [10] Points: X: 253.13148 Y: -414.29672 Z: 44.778492
[0] [11] Points: X: 234.91629 Y: -424.618 Z: 44.611805
[0] [12] Points: X: 223.58096 Y: -426.7876 Z: 44.61183
[0] [13] Points: X: 204.30447 Y: -431.26736 Z: 44.139587
[0] [14] Points: X: 196.25475 Y: -456.6652 Z: 42.611828
[0] [15] Points: X: 223.04488 Y: -479.07944 Z: 32.472946
[0] [16] Points: X: 263.47906 Y: -480.66202 Z: 46.80626
[0] [17] Points: X: 278.39508 Y: -477.34967 Z: 43.88961
[0] [18] Points: X: 276.92447 Y: -457.9421 Z: 37.05632
[0] [19] Points: X: 276.92447 Y: -457.9421 Z: 37.05632
```

Waypoints after areatrigger is reached
```
-- Ghaz'an 
-- Reach areatrigger 4302 -> waypoints
SET @NPC := XXXXXX;
SET @PATH := @NPC * 10;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`visibilityDistanceType`,`auras`) VALUES (@NPC,@PATH,0,0,1,0,0, '8876'); -- Includes sniffed Thrash aura
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,276.92447,-457.9421,37.05632,NULL,0,0,0,100,0),
(@PATH,2,230.84842,-473.81586,44.384487,NULL,0,0,0,100,0),
(@PATH,3,223.15588,-474.34158,46.36878,NULL,0,0,0,100,0),
(@PATH,4,214.68163,-479.58606,52.51196,NULL,0,0,0,100,0),
(@PATH,5,200.88074,-483.32407,58.438652,NULL,0,0,0,100,0),
(@PATH,6,185.70932,-485.64243,64.15632,NULL,0,0,0,100,0),
(@PATH,7,175.2845,-486.21573,67.86781,NULL,0,0,0,100,0),
(@PATH,8,160.93648,-476.7256,74.70425,NULL,0,0,0,100,0),
(@PATH,9,158.37117,-472.06967,74.63962,NULL,0,0,0,100,0),
(@PATH,10,159.37047,-467.53186,75.40111,NULL,0,0,0,100,0),
(@PATH,11,180.78477,-471.30957,77.03179,NULL,0,0,0,100,0),
(@PATH,12,189.26646,-471.74365,77.766594,NULL,0,0,0,100,0),
(@PATH,13,200.31303,-472.0223,79.02495,NULL,0,0,0,100,0),
(@PATH,14,212.294,-480.27985,80.729095,NULL,0,0,0,100,0),
(@PATH,15,222.19577,-476.2322,81.27319,NULL,0,0,0,100,0),
(@PATH,16,233.52731,-471.28235,81.37284,NULL,0,0,0,100,0),
(@PATH,17,239.5655,-468.65033,81.37284,NULL,0,0,0,100,0),
(@PATH,18,248.33444,-465.86627,81.37284,NULL,0,0,0,100,0),
-- (@PATH,19,260.44455,-457.9094,81.37284,NULL,0,0,0,100,0),
-- (@PATH,20,256.2785,-458.73138,81.37284,NULL,0,0,0,100,0),
-- (@PATH,21,256.2785,-458.73138,81.37284,3.543018341064453125,0,0,0,100,0);
-- After that, random movement of range 12
-- Actually only moves up to waypoint 18, I think
```

### AC rev. hash/commit

https://github.com/azerothcore/azerothcore-wotlk/commit/fb08f8e905b92029040017c24198d9e6008a2c5a

### Operating system

Windows 10

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by UltraNix**
Added scripted waypoints.
Fixes #13793

## Issues Addressed:

- Closes #13793

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

`.go xyz 251.88 -434.55 81.372864 546 4.6`
Note where Ghaz'an spawns, see if it is swimming in a circle
go to areatrigger 4302 (`.go xyz 241.09 -373.76 72.447418 546 3.75`)
See if it moves towards the platform
When it reaches the platform, see if it wanders.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Outland/CoilfangReservoir/underbog/instance_the_underbog.cpp [hunk 1]
```diff
@@ -40,6 +33,34 @@ class instance_the_underbog : public InstanceMapScript
...
+        void OnCreatureCreate(Creature* creature) override
+        {
+            switch (creature->GetEntry())
+            {
+                case NPC_GHAZAN:
+                    _ghazanGUID = creature->GetGUID();
+                    break;
+                default:
+                    break;
+            }
```
Comment (Nefertumm) [inline src/server/scripts/Outland/CoilfangReservoir/underbog/instance_the_underbog.cpp:46]: Add InstanceScript::OnCreatureCreate(creature);

**Review by Nefertumm — APPROVED** [review]

**Comment by Gultask** [comment]
I'm not seeing Acid Spit or Tail Lash being cast, only Acid Breath. Tried on both normal and heroic.

**Comment by UltraNix** [comment]
@Gultask Should be okay now.

**Comment by Gultask** [comment]
Works fine now 👍

**Pull request merged**